### PR TITLE
Migration/stavemodifier

### PR DESCRIPTION
**Pull request by rvilarl**
Stavemodifier is there!

### CODE REVIEW

File: src/stavemodifier.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import { Stave } from './stave';
+
+export interface LayoutMetrics {
+  xMin: number;
+  xMax: number;
+  paddingLeft: number;
+  paddingRight: number;
+}
+
+export interface Metrics {
```
Comment (0xfe) [inline src/stavemodifier.ts:17]: I don't think we need this -- this looks like the metrics we use for `StaveNote`

**Comment by rvilarl** [comment]
Ready

**Comment by rvilarl** [comment]
@0xfe anything pending here?

File: src/stave.ts [hunk 2]
```diff
@@ -189,9 +189,7 @@ export class Stave extends Element {
     this.end_x += shift;
     for (let i = 0; i < this.modifiers.length; i++) {
       const mod = this.modifiers[i];
-      if (mod.x !== undefined) {
-        mod.x += shift;
-      }
+      mod.setX(mod.getX() + shift);
```
Comment (0xfe) [inline src/stave.ts:192]: What if `mod.x` is `undefined`? Seems like we should catch that and throw an exception earlier on.

File: src/stavemodifier.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
+// [VexFlow](http://vexflow.com) - Copyright (c) Mohit Muthanna 2010.
+//
+// ## Description
+// A base class for stave modifiers (e.g. clefs, key signatures)
+
+import { Element } from './element';
+import { Glyph } from './glyph';
+import { Stave } from './stave';
+import { Metrics } from './note';
+export interface LayoutMetrics {
```
Comment (0xfe) [inline src/stavemodifier.ts:10]: Newline before export please.

File: src/stavemodifier.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+  setX(x: number): this {
+    this.x = x;
+    return this;
+  }
+
+  getCategory(): string {
+    return '';
+  }
+
+  makeSpacer(padding: number): StaveModifierSpacer {
```
Comment (0xfe) [inline src/stavemodifier.ts:93]: Let's remove this  method and the `StaveModifierSpacer` -- it's not used anywhere.

File: src/stavemodifier.ts (hunk 3 above)
Comment (rvilarl) [reply src/stavemodifier.ts:10]: done

File: src/stavemodifier.ts (hunk 4 above)
Comment (rvilarl) [reply src/stavemodifier.ts:93]: done

File: src/stave.ts (hunk 2 above)
Comment (rvilarl) [reply src/stave.ts:192]: mod.getX() returns a number, undefined is not possible.

**Comment by rvilarl** [comment]
Ready, hopefully :)

**Comment by 0xfe** [comment]
Looks good.

**Pull request merged**